runners/flink/src/main/java/org/apache/beam/runners/flink/adapter/FlinkInput.java

```diff
@@ -71,7 +71,7 @@ public String getUrn() {
     }
 
     @Override
-    @SuppressWarnings("nullness")  // TODO(https://github.com/apache/beam/issues/20497)
+    @SuppressWarnings("nullness") // TODO(https://github.com/apache/beam/issues/20497)
```

Comment:
Just curious but what can cause an NPE in this code?

Reply:
This isn't about a NPE that could happen, but a type mismatch in nested wildcard template parameters (if I understood the error correctly).

Reply:
OK that sounds like a pain and I'm OK with it. Can you narrow the suppression to one line in the body of the method instead of the whole method? (per rationale in other comment)

Reply:
That's not possible because the problem is the type checking of the overridden method signature. (This will require a change to the subclass and all its implementations, which may be good to do but out of scope.)

Reply:
Ah that's a bummer.

... But wait this is actually our own code and the nullness in the types is as intended. The return value is optional. https://github.com/apache/beam/blob/2f3fac6e14017f8d1d2d6cae42c6c3366d9065bb/sdks/java/core/src/main/java/org/apache/beam/sdk/util/construction/PTransformTranslation.java#L620

Reply:
It wasn't the return type that was causing issues.

Reply:
Yea ha I cloned it and checked. Of course, the return type is also mismatched and it is an example of suppression overshadowing. But wow yea I fully understand taking a pass on fixing that wild generics / variance issue.